gh-130160: use `.. program::` directive for documenting `idle` CLI (#130278)
python/cpython: Doc/library/idle.rst

Command line usage
^^^^^^^^^^^^^^^^^^

.. program:: idle

IDLE can be invoked from the command line with various options. The general syntax is:

.. code-block:: bash

   python -m idlelib [options] [file ...]

The following options are available:

.. option:: -c <command>

>    Run the specified Python command in the shell window. For example, pass ``-c "print('Hello, World!')"``.
> On Windows, the outer quotes must be double quotes as shown.

Reviewer: Try with indent.
```suggestion
   Run the specified Python command in the shell window. For example, pass ``-c "print('Hello, World!')"``.
   On Windows, the outer quotes must be double quotes as shown.
```

>    Run the specified Python command in the shell window. For example, pass ``-c "print('Hello, World!')"``.
>    On Windows, the outer quotes must be double quotes as shown.

Reviewer: ```suggestion
   Run the specified Python command in the shell window. For example, pass ``-c "print('Hello, World!')"``.\
   On Windows, the outer quotes must be double quotes as shown.
```

>    On Windows, the outer quotes must be double quotes as shown.

Reviewer: @picnixz I want to be able to make comments more than 1 line long, such as here, without having super long lines in the .rst file.  Quick reference [directives](https://devguide.python.org/documentation/markup/#directives) just says "The directive content follows after a blank line and is indented relative to the directive start."  Unindenting the line is an error and adding `\` to the previous line does nothing.  All indented lines are supposed to be part of the directive, but the above seems to handle "On Windows ..." otherwise.  It is printed unindented after a blank line.

Reviewer 2: A newline should work? Sphinx/DocUtils only creates a new paragraph on a blank line (\n\n).
```suggestion
   Run the specified Python command in the shell window.
   For example, pass ``-c "print('Hello, World!')"``.
   On Windows, the outer quotes must be double quotes as shown.
```

Reviewer: I asked because the simple newline I had did not work.  But two days later, it does.  But I like the 2nd line split and merged it.

>    Run the specified Python command in the shell window. For example, pass ``-c "print('Hello, World!')"``.\
>    On Windows, the outer quotes must be double quotes as shown.

Reviewer: Remove backslash.
```suggestion
   Run the specified Python command in the shell window. For example, pass ``-c "print('Hello, World!')"``.
   On Windows, the outer quotes must be double quotes as shown.
```

.. option:: -d

   Enable the debugger and open the shell window.

.. option:: -e

   Open an editor window.

.. option:: -h

   Print a help message with legal combinations of options and exit.

.. option:: -i

   Open a shell window.

.. option:: -r <file>

   Run the specified file in the shell window.

.. option:: -s

   Run the startup file (as defined by the environment variables :envvar:`IDLESTARTUP` or :envvar:`PYTHONSTARTUP`) before opening the shell window.

.. option:: -t <title>

   Set the title of the shell window.

.. option:: -

   Read and execute standard input in the shell window. This option must be the last one before any arguments.

If arguments are provided:

> - If ``-``, ``-c``, or ``-r`` is used, all arguments are placed in ``sys.argv[1:]``, and ``sys.argv[0]`` is set to ``''``, ``'-c'``, or ``'-r'``. No editor window is opened, even if that is the default set in the Options dialog.

Reviewer 3: ```suggestion
- If ``-``, ``-c``, or ``-r`` is used, all arguments are placed in ``sys.argv[1:]``,
  and ``sys.argv[0]`` is set to ``''``, ``'-c'``, or ``'-r'`` respectively.
  No editor window is opened, even if that is the default set in the *Options* dialog.
```

I think emphasising the "Options" word would perhaps render a bit better?

Author: I agree - emphasizing "Options" would make it stand out better.

Command line usage
^^^^^^^^^^^^^^^^^^

.. program:: idle

IDLE can be invoked from the command line with various options. The general syntax is:

.. code-block:: bash

>    python -m idlelib idle [options] [file ...]

Reviewer 4: ```suggestion
   python -m idlelib [options] [file ...]
```

Small error

Author: Sorry will resolve it right now

Author: Sorry will resolve it right now

Reviewer 4: If you are using the GitHub app/website you can apply suggestions as a batch, sometimes it’s a bit easier than opening up an ide :-)

The following options are available:

.. option:: -c <command>

>    Run the specified Python command in the shell window. For example, pass ``-c "print('Hello, World!')"``.
> 
> On Windows, the outer quotes must be double quotes as shown.

Reviewer: A blank line does not work.
```suggestion
   Run the specified Python command in the shell window. For example, pass ``-c "print('Hello, World!')"``.
On Windows, the outer quotes must be double quotes as shown.
```

Command line usage
^^^^^^^^^^^^^^^^^^

.. program:: idle

IDLE can be invoked from the command line with various options. The general syntax is:

.. code-block:: bash

>    idle [options] [file ...]

Reviewer: ```suggestion
   python -m idlelib [options] [file ...]
```
Same as on dis page.

The following options are available:

.. option:: -c <command>

>    Run the specified Python command in the shell window. For example:
> 
>    .. code-block:: bash
> 
>       idle -c "print('Hello, World!')"

Reviewer: ```suggestion
   Run the specified Python command in the shell window. For example, pass ``-c "print('Hello, World!')"``.
On Windows, the outer quotes must be double quotes as shown.  
```
Can we break the line?  If so, skip indent on next line as above?  Some other lines below are over 80 chars.

Reviewer: There was no blank line between the two + lines in the suggestion.  I removed it now and am retesting.
